cmd/system-probe/modules/gpu.go

```diff
@@ -147,3 +148,49 @@ func hostRoot() string {
 
 	return "/"
 }
+
+var agentProcessRegexp = regexp.MustCompile("datadog-agent/.*/agent")
+
+func getAgentPID(procRoot string) (uint32, error) {
+	pids, err := kernel.AllPidsProcs(procRoot)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get all pids: %w", err)
+	}
+
+	for _, pid := range pids {
+		proc := uprobes.NewProcInfo(procRoot, uint32(pid))
+		exe, err := proc.Exe()
+		if err != nil {
+			// Ignore this process, we don't want to stop the search because of that
+			continue
+		}
+
+		if agentProcessRegexp.MatchString(exe) {
+			return uint32(pid), nil
+		}
+	}
+
+	return 0, errors.New("agent process not found")
+}
+
+func configureCgroupPermissions() {
```

Comment:
can you add inline comments in this function to explain we need to adjust the device permissions for both agent and system-probe (and briefly explain why).